test/gtest/uct/ib/test_sockaddr.cc

```diff
@@ -885,27 +893,59 @@ UCS_TEST_P(test_uct_cm_sockaddr, connect_client_to_server_reject_with_delay)
     test_delayed_server_response(true);
 }
 
-UCS_TEST_P(test_uct_cm_sockaddr, double_ep_disconnect)
+UCS_TEST_P(test_uct_cm_sockaddr, ep_disconnect_err_codes)
 {
-    ucs_status_t status;
+    bool disconnecting = false;
 
     cm_listen_and_connect();
 
+    m_client->async_block();
+        if (m_cm_state & TEST_CM_STATE_CLIENT_CONNECTED) {
+            UCS_TEST_MESSAGE << "EXP: " << ucs_status_string(UCS_OK);
+            EXPECT_EQ(UCS_OK, uct_ep_disconnect(m_client->ep(0), 0));
+            disconnecting = true;
+        } else {
+            UCS_TEST_MESSAGE << "EXP: " << ucs_status_string(UCS_ERR_BUSY);
+            EXPECT_EQ(UCS_ERR_BUSY, uct_ep_disconnect(m_client->ep(0), 0));
+        }
+    m_client->async_unblock();
+
     wait_for_bits(&m_cm_state, TEST_CM_STATE_SERVER_CONNECTED |
                                TEST_CM_STATE_CLIENT_CONNECTED);
     EXPECT_TRUE(ucs_test_all_flags(m_cm_state, (TEST_CM_STATE_SERVER_CONNECTED |
                                                 TEST_CM_STATE_CLIENT_CONNECTED)));
 
-    ASSERT_UCS_OK(uct_ep_disconnect(m_client->ep(0), 0));
+    m_client->async_block();
```

Comment:
can the test use its own lock mechanism instead of async block/unblock

Reply:
then need to add additional lock/unlock to callbacks, it will affect other tests which don't require it. I use async_block to prevent callback invokavion from UCT in this section. IMO it makes sense in this case, WDYT?

Reply:
ok, so should remove m_lock?
also better add class scoped_aync_block instead of block/unblock wrappers

Reply:
yep, m_lock is a leftover...
> also better add class scoped_aync_block instead of block/unblock wrappers

ok